pkg/gatewayserver/io/basicstationlns/basicstationlns.go

```diff
@@ -137,6 +137,7 @@ func (s *srv) handleDiscover(c echo.Context) error {
 	}
 	ctx, ids, err = s.server.FillGatewayContext(ctx, ids)
 	if err != nil {
+		ctx = log.NewContext(c.Request().Context(), logger)
 		logger.WithError(err).Warn("Failed to fetch gateway")
 		writeDiscoverError(ctx, ws, fmt.Sprintf("Failed to fetch gateway: %s", err.Error()))
```

Comment:
I think we shouldn't rely on the returned `ctx` if there's an error, especially if the method is called `FillGatewayContext`.

I would instead declare `filledCtx` only assign `ctx` when there's no error. In the error handling, use the original `ctx`